--- a/src/tusbd/cdev.c
+++ b/src/tusbd/cdev.c
@@ -10,6 +10,12 @@
 #include <linux/fs.h>
 #include <linux/uaccess.h>
 #include <linux/slab.h>
+#include <linux/version.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,4,0)
+#define class_create(owner, name) class_create(name)
+#define DEV_UEVENT_CONST const
+#else
+#define DEV_UEVENT_CONST
+#endif
 
 /* Original device uevent function */
-static int dev_uevent(struct device *dev, struct kobj_uevent_env *env)
+static int dev_uevent(DEV_UEVENT_CONST struct device *dev, struct kobj_uevent_env *env)
 {
     add_uevent_var(env,"MODALIAS=usb-redirector");
     return 0;
 }
