--- a/src/tusbd/cdev.c
+++ b/src/tusbd/cdev.c
@@ -9,6 +9,7 @@
 
 #include <linux/device.h>
 #include <linux/mutex.h>
+#include <linux/version.h>
 #include <linux/cdev.h>
 #include "usbd.h"
 #include "cdev.h"
@@ -40,7 +41,11 @@
 );
 do{
 mutex_init(&lIIIlIl);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,4,0)
+lIIIlII=class_create(IIIllII);
+#else
 lIIIlII=class_create(THIS_MODULE,IIIllII);
+#endif
 if(lIIIlII==
 
 NULL||IS_ERR(lIIIlII)){lIlIlI(
